connector/mqtt/vernemq/broker/vernemq-1.10.0-fingerprint.patch

```diff
@@ -0,0 +1,82 @@
+diff -Naur vernemq-1.10.0/apps/vmq_server/src/vmq_ssl.erl vernemq-1.10.0-cert-fingerprint/apps/vmq_server/src/vmq_ssl.erl
+--- vernemq-1.10.0/apps/vmq_server/src/vmq_ssl.erl	2019-11-26 07:11:10.000000000 -0300
++++ vernemq-1.10.0-cert-fingerprint/apps/vmq_server/src/vmq_ssl.erl	2021-07-12 11:20:59.282576280 -0300
+@@ -18,15 +18,35 @@
+          cert_to_common_name/1,
+          opts/1]).
+ 
++-define(CERTIFICATE_ACL_URL, os:getenv("CERTIFICATE_ACL_URL", "http://certificate-acl:3000/internal/api/v1/acl-entries/")).
++-define(CERTIFICATE_ACL_REQ_TIMEOUT_MS, erlang:list_to_integer(os:getenv("CERTIFICATE_ACL_REQ_TIMEOUT_MS", "1000"), 10)).
++
++get_acl_entry(Fingerprint) ->
++    Url = string:concat(?CERTIFICATE_ACL_URL, Fingerprint),
++    HTTPOpts =  [{timeout, ?CERTIFICATE_ACL_REQ_TIMEOUT_MS}],
++    case httpc:request(get, {Url, []}, HTTPOpts, []) of
++        {ok, {{_, 200, _}, _, Body}} ->
++            % the body is a json string \"entry\"
++            list_to_binary(string:sub_string(Body, 2, length(Body)-1));
++        _  ->
++            {error, acl_failure}
++    end.
++
+ socket_to_common_name(Socket) ->
+     case ssl:peercert(Socket) of
+         {error, no_peercert} ->
+             undefined;
+         {ok, Cert} ->
+-            OTPCert = public_key:pkix_decode_cert(Cert, otp),
+-            TBSCert = OTPCert#'OTPCertificate'.tbsCertificate,
+-            Subject = TBSCert#'OTPTBSCertificate'.subject,
+-            extract_cn(Subject)
++            Fingerprint = crypto:hash(sha256, Cert),
++
++            % the fingerprint at this point is with formatted but with the ":" as it's last character
```

Comment:
Just sounded like two typos, fixing: "the fingerprint at this point is formatted but with the ":" as its last character"

Reply:
I agree with you, but I won't change it right now. This is not new code, I only changed the way this patch is applied. I think we'll need to refactor this code at some moment and we can address this comment when that happens.